src/utils.cpp

```diff
@@ -583,6 +583,36 @@ std::string utils::get_potential_units(const std::string & path, const std::stri
   return "";
 }
 
+/* ----------------------------------------------------------------------
+   return bitmask of supported conversions for a given property
+------------------------------------------------------------------------- */
+int utils::get_supported_conversions(const int property)
+{
+  if (property == ENERGY) {
+    return METAL2REAL | REAL2METAL;
+  }
+  return NOCONVERT;
+}
+
+/* ----------------------------------------------------------------------
+   return conversion factor for a given property and conversion setting
+   return 0.0 if unknown.
+------------------------------------------------------------------------- */
+
+double utils::get_conversion_factor(const int property, const int conversion)
+{
+  if (property == ENERGY) {
+    if (conversion == NOCONVERT) {
+      return 1.0;
+    } else if (conversion == METAL2REAL) {
+      return 23.060549;
```

Comment:
how were these computed, why is the accuracy of one so much higher than the other?  Can these be computed from the factors already defined in update.cpp ?

Reply:
this *is* `force->qe2f` for "real" units. **accuracy** of this number is rather low (when compared with the most recent CODATA numbers), the inverse has more *precision* so that the back and forth conversion matches to reasonably high precision (about 1.0e-12 for most cases but i am currently only checking update 1.0e-8). Also, i noticed, that in general, there is limited consistency for the conversions because the factors in update.cpp are not very precise. for example, comparison of pressure (which is in atm vs. bar) can only be matched to about 1.0e-8 relative precision due to the precision of the factors for computing those.